### PR TITLE
Update log BBE

### PR DESCRIPTION
**Pull request by MadhukaHarith92**

### CODE REVIEW

File: examples/log-api/log_api.bal [hunk 1]
```diff
@@ -4,13 +4,19 @@ public function main() {
     error e = error("something went wrong!");
 
     // There are two log levels (INFO, ERROR) and there is no user configuration to control which log level to print.
-    // Users can pass any number of key/value pairs which needs to be displayed in the log message.
     log:print("info log");
-    log:print("info log", id = 845315, name = "foo");
+    log:printError("error log");
+
+    // Users can pass any number of key/value pairs which needs to be displayed in the log message.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:10]:
```suggestion
    // You can pass any number of key/value pairs, which need to be displayed in the log message.
```

File: examples/log-api/log_api.bal [hunk 2]
```diff
@@ -4,13 +4,19 @@ public function main() {
     error e = error("something went wrong!");
 
     // There are two log levels (INFO, ERROR) and there is no user configuration to control which log level to print.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:6]:
```suggestion
    // There are two log levels (`INFO` and `ERROR`) and there is no user configuration to control which log level to print.
```

File: examples/log-api/log_api.bal [hunk 3]
```diff
@@ -4,13 +4,19 @@ public function main() {
...
-    // Users can pass any number of key/value pairs which needs to be displayed in the log message.
     log:print("info log");
-    log:print("info log", id = 845315, name = "foo");
+    log:printError("error log");
+
+    // Users can pass any number of key/value pairs which needs to be displayed in the log message.
+    // These can be of anydata type including int, string and boolean.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:11]:
```suggestion
    // These can be of any data type including int, string, and boolean.
```

File: examples/log-api/log_api.bal [hunk 4]
```diff
@@ -4,13 +4,19 @@ public function main() {
...
+    log:print("info log", id = 845315, name = "foo", successful = true);
+
+    // Users can also pass key/value pairs where the values are function pointers.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:14]:
```suggestion
    // You can also pass key/value pairs in which the values are function pointers.
```

File: examples/log-api/log_api.bal [hunk 5]
```diff
@@ -4,13 +4,19 @@ public function main() {
...
     log:print("info log", id = 845315,
               name = isolated function() returns string { return "foo";});
-    log:printError("error log");
     log:printError("error log",
         id = isolated function() returns int { return 845315;}, name = "foo");
+
+    // Optionally an error can be passed to the printError function.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:20]:
```suggestion
    // Optionally, an error can be passed to the `printError` function.
```

File: examples/log-api/log_api.bal (hunk 3 above)
Comment (MadhukaHarith92) [reply examples/log-api/log_api.bal:11]: IINM isn't "any data" type different to `anydata` type in Ballerina.

Comment (daneshk) [reply examples/log-api/log_api.bal:11]: Shall we add backtick(`) for anydata

Comment (MadhukaHarith92) [reply examples/log-api/log_api.bal]: Updated in 68a609c

Comment (praneesha) [reply examples/log-api/log_api.bal]: +1

**Review by praneesha — APPROVED** [review]

**Pull request merged**